ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Eras.hs

```diff
@@ -331,6 +335,24 @@ instance SupportsTwoPhaseValidation (BabbageEra c) where
         ) -> Just claimedFlag
     _ -> Nothing
 
+instance SupportsTwoPhaseValidation (ConwayEra c) where
+  incorrectClaimedFlag _ pf = case pf of
+    SL.UtxowFailure
+      ( Babbage.AlonzoInBabbageUtxowPredFailure
```

Comment:
Conway reuses Babbage's UTXO rules. So the `AlonzoInBabbageUtxowPredFailure` name is right, but confusing here.

Reply:
I chatted with ledger about this. There's no clear improvement to be made. The reasoning:

- Conway is actually defined to use rules that were introduced for Babbage and hence _named_ after Babbage.
- This constructor is from those Babbage rules, so it's embedding an error from the Alonzo rules into the errors for the Babbage rules.
- Hence it's name is correct, and not so confusing, if someone understands that the Conway era directly reuses the Babbage rules.